src/linear_algebra/multilinear.lean

```diff
@@ -530,6 +530,22 @@ end semimodule
 
 section
 
+/-- Apply a permutation to the order of the arguments, obtaining another multilinear map.
+
+The naming is derived from `finsupp.dom_congr`, noting that here the permutation applies to the
+domain of the domain. -/
+@[simps apply]
+def dom_dom_congr [add_comm_monoid M₃] [semimodule R M₃]
+  (m : multilinear_map R (λ i : ι, M₂) M₃) (σ : equiv.perm ι) :
+  multilinear_map R (λ i : ι, M₂) M₃ :=
```

Comment:
In hindsight, this should have had signature
```suggestion
  (m : multilinear_map R (λ i : ι1, M₂) M₃) (σ : ι1 ≃ ι2) :
  multilinear_map R (λ i : ι2, M₂) M₃ :=
```

I'll make a follow-up PR to generalize this, since I don't want to have to build all the way from `function.basic` again! Let's let bors finish merging this one.

Reply:
I think this should perhaps be:
```
@[simps apply]
private def dom_dom_congr_aux {M₂ M₃ : Type*} {ι₁ ι₂ : Type*} [decidable_eq ι₁] [decidable_eq ι₂]
  [add_comm_monoid M₂] [add_comm_monoid M₃] [semimodule R M₂] [semimodule R M₃]
  (σ : ι₁ ≃ ι₂) (m : multilinear_map R (λ i : ι₁, M₂) M₃) : multilinear_map R (λ i : ι₂, M₂) M₃ :=
{ to_fun := λ v, m (λ i, v (σ i)),
  map_add' := λ v i a b, by { simp_rw function.update_apply_equiv_apply v, rw m.map_add, },
  map_smul' := λ v i a b, by { simp_rw function.update_apply_equiv_apply v, rw m.map_smul, }, }

/-- Transfer the equivalence between argument indices to an equivalence between maps
The naming is derived from `finsupp.dom_congr`, noting that here the permutation applies to the
domain of the domain. -/
@[simps apply symm_apply]
def dom_dom_congr {M₂ M₃ : Type*} {ι₁ ι₂ : Type*} [decidable_eq ι₁] [decidable_eq ι₂]
  [add_comm_monoid M₂] [add_comm_monoid M₃] [semimodule R M₂] [semimodule R M₃]
  (σ : ι₁ ≃ ι₂) :
  multilinear_map R (λ i : ι₁, M₂) M₃ ≃+ multilinear_map R (λ i : ι₂, M₂) M₃ :=
{ to_fun := dom_dom_congr_aux σ,
  inv_fun := dom_dom_congr_aux σ.symm,
  left_inv := λ m, by {ext, simp},
  right_inv := λ m, by {ext, simp},
  map_add' := λ a b, by {ext, simp} }
```

Reply:
#5180